ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -0,0 +1,85 @@
+try:
+    from tqdm import tqdm
+except ImportError:
+    raise RuntimeError("This contrib module requires tqdm to be installed")
+
+from ignite.engine import Events
+
+
+class ProgressBar:
+    """
+    TQDM progress bar handler to log training progress and computed metrics
+
+    Examples:
+
+    .. code-block:: python
+
+        pbar = ProgressBar()
+        pbar.attach(trainer, len(data_loader), ['loss'], mode='iteration', log_interval=1)
+
+    Note:
+        When adding this handler to an engine, it is recommend that you replace every print
+        operation in the engine's handlers with `tqdm.write()` to guarantee the correct stdout format.
+    """
+
+    def __init__(self):
+        self.pbar = None
+
+    def _reset(self, engine, num_iterations):
+        self.pbar = tqdm(
+            total=num_iterations,
+            leave=False,
+            bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
+
+    def _close(self, engine):
+        self.pbar.close()
+
+    @staticmethod
+    def _log_message(engine, metric_names, mode, log_interval):
+
+        i = engine.state.epoch if mode == 'epoch' else engine.state.iteration
+
+        if log_interval and i % log_interval == 0:
+            if mode == 'epoch':
+                message = 'Epoch {}'.format(engine.state.epoch)
+            else:
+                message = 'Iteration {}'.format(engine.state.iteration)
+
+            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            for name, value in metrics.items():
+                message += ' | {}={}'.format(name, value)
+
+            tqdm.write(message)
+
+    def _update(self, engine, metric_names):
+        metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+        self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
+        self.pbar.set_postfix(**metrics)
+        self.pbar.update()
+
+    def attach(self, engine, num_iterations, metric_names, mode='epoch', log_interval=1):
```

Comment:
Just wonder what happens if user attaches to multiple engines:
```python
pbar = ProgressBar()
pbar.attach(trainer1, len(data_loader1), ['loss'], mode='iteration', log_interval=1)
pbar.attach(trainer2, len(data_loader2), ['loss'], mode='iteration', log_interval=1)
```
maybe we need two tqdm progress bars ?

Reply:
If they are running concurrently, we would need two. If not, there would be no problem.